### PR TITLE
New model configuration context

### PR DESCRIPTION
**Pull request by brandomr**
# Overview
This PR adds a new context called `model_configuration` that is designed to support flexible edits to Terarium's new model configuration schema.

At run time, it uses Terarium's swagger spec to "learn" the schema of model configurations. It also takes in a model configuration `id` and then enables the user to ask questions of the configuration or make updates to it. In this fashion, the agent is flexible and robust to future changes to the model configuration schema and consistently writes edits to the configuration that are schema compliant. Note that no schema validation is in place after the agent edits the configuration, but this is possible if we'd like.

## Capabilities

The configuration is loaded as a Python dictionary called `model_config`:

Screenshot 2024-07-02 at 3 29 28 PM

The user can ask the agent basic things about the configuration:
Screenshot 2024-07-02 at 3 29 51 PM

The user can ask the agent to edit the config:
Screenshot 2024-07-02 at 3 30 51 PM

The user can ask for edits to things besides just parameters (e.g. initials, observables):
Screenshot 2024-07-02 at 3 36 38 PM

The user can ask for information about the schema and what is allowed:
Screenshot 2024-07-02 at 3 31 08 PM

### CODE REVIEW

**Review by blanchco — APPROVED** [review]
Tested it out and looks good we'll have to make some changes on our side for the model config drilldown to use the new context 👍

**Review by dvince2 — APPROVED** [review]

File: src/askem_beaker/contexts/model_configuration/procedures/python3/get_schema.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import requests
+import json
+import os
+
+# Fetch the Swagger spec
+url = f"{os.environ['HMI_SERVER_URL']}/v3/api-docs"
```
Comment (mattprintz) [inline src/askem_beaker/contexts/model_configuration/procedures/python3/get_schema.py:6]: If we want to future-proof, env variables should be collected in the context.py file passed in as a template render argument as we cannot guarantee that the procedure will run in the same environment as the beaker kernel.

Comment (brandomr) [reply src/askem_beaker/contexts/model_configuration/procedures/python3/get_schema.py:6]: Good catch I'll address that

Comment (brandomr) [reply src/askem_beaker/contexts/model_configuration/procedures/python3/get_schema.py]: Fixed that

**Review by mattprintz — APPROVED** [review]

**Pull request merged**